waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/tunnelling/TunnelingMetaStoreClientFactory.java

```diff
@@ -41,55 +38,65 @@
   @VisibleForTesting
   final MethodChecker METHOD_CHECKER = new MetastoreClientMethodChecker();
 
-  private final TunnelableFactorySupplier tunnelableFactorySupplier;
+  private HiveConf localHiveConf;
+  private final TunnelableFactory<CloseableThriftHiveMetastoreIface> tunnelableFactory;
+  private final String localhost;
   private final MetaStoreClientFactory defaultFactory;
-  private final LocalHiveConfFactory localHiveConfFactory;
   private final HiveMetaStoreClientSupplierFactory hiveMetaStoreClientSupplierFactory;
 
   public TunnelingMetaStoreClientFactory() {
-    this(new TunnelableFactorySupplier(), new DefaultMetaStoreClientFactory(), new LocalHiveConfFactory(),
-        new HiveMetaStoreClientSupplierFactory());
+    this(new DefaultMetaStoreClientFactory(), new HiveMetaStoreClientSupplierFactory(), null, null);
```

Comment:
we use a `DefaultMetaStoreClientFactory` to fall back on, but we also do that in the helper, we should probably remove it here so this only deals with tunnels

Reply:
So instead of creating a new `DefaultMetaStoreClientFactory` when there is no `tunnelableFactory` (line 71), I could just return an exception, right?

Reply:
So instead of creating a new `DefaultMetaStoreClientFactory` when there is no `tunnelableFactory` (line 71), I could just return an exception, right?